krm-functions/lib/interface/v1alpha1/interface_test.go

```diff
@@ -0,0 +1,440 @@
+		errExpected           bool
+		defaultCNIType        string
+		defaultAttachmentType string
+	}{
+		"SetInterfaceSpecNormal": {
+			file:                  itface,
+			defaultCNIType:        "sriov",
+			defaultAttachmentType: "vlan",
+			spec: &nephioreqv1alpha1.InterfaceSpec{
+				NetworkInstance: &corev1.ObjectReference{
+					Name: "test",
+				},
+				AttachmentType: nephioreqv1alpha1.AttachmentTypeNone,
+				CNIType:        nephioreqv1alpha1.CNITypeIPVLAN,
+			},
+			errExpected: false,
+		},
+		"SetInterfaceSpecDefault": {
+			file:                  itface,
+			defaultCNIType:        "",
+			defaultAttachmentType: "",
+			spec: &nephioreqv1alpha1.InterfaceSpec{
+				NetworkInstance: &corev1.ObjectReference{
+					Name: "test",
+				},
+			},
+			errExpected: false,
+		},
+		"SetInterfaceSpecEmpty": {
+			file:                  itfaceEmpty,
+			defaultCNIType:        "",
+			defaultAttachmentType: "",
+			spec: &nephioreqv1alpha1.InterfaceSpec{
+				NetworkInstance: &corev1.ObjectReference{
+					Name: "test",
+				},
+			},
+			errExpected: false,
+		},
+	}
+
+	for name, tc := range cases {
+		i, err := NewFromYAML([]byte(tc.file))
+		if err != nil {
+			t.Errorf("cannot unmarshal file: %s", err.Error())
+		}
+
+		t.Run(name, func(t *testing.T) {
+			err := i.SetSpec(tc.spec)
+			if tc.errExpected {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				if tc.spec.NetworkInstance != nil {
+					got := i.GetNetworkInstanceName()
+					if diff := cmp.Diff(tc.spec.NetworkInstance.Name, got); diff != "" {
+						t.Errorf("TestSetInterfaceSpec: -want, +got:\n%s", diff)
+					}
+				}
+				if tc.spec.AttachmentType != "" {
+					got := i.GetAttachmentType()
+					if diff := cmp.Diff(string(tc.spec.AttachmentType), got); diff != "" {
+						t.Errorf("TestSetInterfaceSpec: -want, +got:\n%s", diff)
+					}
+				} else {
+					got := i.GetAttachmentType()
+					if diff := cmp.Diff(tc.defaultAttachmentType, got); diff != "" {
+						t.Errorf("TestSetInterfaceSpec: -want, +got:\n%s", diff)
+					}
+				}
+				if tc.spec.CNIType != "" {
+					got := i.GetCNIType()
+					if diff := cmp.Diff(string(tc.spec.CNIType), got); diff != "" {
+						t.Errorf("TestSetInterfaceSpec: -want, +got:\n%s", diff)
+					}
+				} else {
+					got := i.GetCNIType()
+					if diff := cmp.Diff(tc.defaultCNIType, got); diff != "" {
+						t.Errorf("TestSetInterfaceSpec: -want, +got:\n%s", diff)
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestDeleteCNIType(t *testing.T) {
+	cases := map[string]struct {
+		file string
+	}{
+		"DeleteCNIType": {
+			file: itface,
+		},
+		"DeleteCNITypeEmpty": {
+			file: itfaceEmpty,
+		},
+	}
+
+	for name, tc := range cases {
+		i, err := NewFromYAML([]byte(tc.file))
+		if err != nil {
+			t.Errorf("cannot unmarshal file: %s", err.Error())
+		}
+
+		t.Run(name, func(t *testing.T) {
+			err := i.DeleteCNIType()
+			assert.NoError(t, err)
+
+			fmt.Println(i.GetKubeObject().String())
+		})
+	}
+}
```

Comment:
May be missing test for DeleteAttachmentType.

Reply:
added